src/main/scala/edu/knowitall/srlie/SrlExtraction.scala

```diff
@@ -222,6 +252,11 @@ object SrlExtraction {
       )
       case _ => false
     }
+
+    override def hashCode(): Int = {
+      val state = Seq(super.hashCode(), text, tokens, intervals)
```

Comment:
I wouldn't bother including the super hash code, since you're already using all of the equality fields.